src/main/webapp/profileScript.js

```diff
@@ -78,3 +81,39 @@ async function addLogoutLink() {
 
   document.getElementById("logout-link").href = loginInfo.actionUrl;
 }
+
+// Load all routes for the logged user.
+function loadRoutes() {
+  fetch("/user-routes")
+    .then((response) => response.json())
+    .then((routesList) => {
+      for (i in routesList) {
+        addRoute(routesList[i]);
+      }
+    });
+}
+
+function addRoute(newRoute) {
+  let card = document.createElement("div");
+  let container = document.createElement("div");
+  let routeDetails = document.createElement("div");
+  let routeImg = document.createElement("img");
+  let routeRating = document.createElement("div");
+
+  card.classList.add("card");
+  container.classList.add("container");
+  routeDetails.classList.add("route-details");
+  routeImg.classList.add("route-img");
+  routeRating.classList.add("rating");
+
+  routeDetails.innerHTML = newRoute["routeName"];
+  let emptyStart = '<span class="far fa-star"></span>';
+  routeRating.innerHTML = emptyStart.repeat(5);
+
+  card.appendChild(container);
+  container.appendChild(routeDetails);
+  container.appendChild(routeImg);
+  container.appendChild(routeRating);
+
+  document.getElementById("completed-routes").appendChild(card);
```

Comment:
I think it would make more sense to add these routes to the `Future routes` section and to create some kind of `Mark as completed` button for them

Reply:
I wanted to add new property for that in the next PR along with other features like image and rating 🤔